### ISSUE TITLE
Test for multiple assumptions to see if the env::verify function does automatic matching

### ISSUE THREAD

**Issue by ercecan** [chainwayxyz/citrea#1401]
For guest side verification we send assumptions from the host to the guest which are basically receipts of batch proofs in our case. To verify them in guest we also send their corresponding journals. In some cases there will be more than one batch proof to verify. To make sure the env::verify function does match the journals to the assumptions we should write a test with multiple batch proofs landing an L1 block and the light client prover should successfully verify and create light client proof for these batch proofs 

**TLDR**: Test for multiple assumptions to see if the env::verify function does automatic matching between the journal and the assumption or do we need to verify them in order?

**Comment by eyusufatik** [comment]
@ozankaymak confirmed this works
